src/main/java/org/ossgang/commons/mapbackeds/Mapbackeds.java

```diff
@@ -157,23 +245,40 @@ public M build() {
     }
 
     public static Set<Method> fieldMethods(Class<?> intfc) {
+        return collectIntfcMethods(intfc, m -> !m.isDefault());
+    }
+
+    public static Optional<Method> toStringMethod(Class<?> intfc) {
+        Set<Method> toStringMethods = toStringMethods(intfc);
+        if (toStringMethods.size() > 1) {
+            throw new IllegalArgumentException(
+                    "More than one method with a ToString annotation found in the hierarchy!");
+        }
+        return toStringMethods.stream().findFirst();
+    }
+
+    private static Set<Method> toStringMethods(Class<?> intfc) {
+        return collectIntfcMethods(intfc, m -> m.isAnnotationPresent(ToString.class));
+    }
+
+    private static Set<Method> collectIntfcMethods(Class<?> intfc, Predicate<Method> methodPredicate) {
         requireInterface(intfc);
-        Set<Method> fields = fieldsFrom(intfc);
+        Set<Method> fields = findZeroParamMethodsFrom(intfc, methodPredicate);
         Class<?>[] superinterfaces = intfc.getInterfaces();
         for (Class<?> supi : superinterfaces) {
-            fields.addAll(fieldMethods(supi));
+            fields.addAll(collectIntfcMethods(supi, methodPredicate));
```

Comment:
Is this really correct? Should @ToString methods be inherited by subinterfaces? It is kind of the same dilemma as with toString() methods on objects - sometimes it may be handy to inherit them, but on the other hand, in most cases they will not do the full job, as the subinterfaces most likely will contain some extra methods/fields.

What if ...
```java
interface A {
@ToString
String asString();
}

interface B extends A {
@ToString
@Override
String asString();
}
```
... will this be handled?

Reply:
Mhmm.... indeed it is a good question ... As this mechanism is at the moment used for some kind of domain objects, I think it is useful. Having two is anyhow not allowed... 

Indeed I did not test scenarios where the methods override each other ...  but on the other hand: They should fall in one of the cases:
a) being a 'field' method, thus being contained in the map
b) being a default method, thus being executed on the proxy.... 

I think, I will test a bit more ...

Reply:
Sorry, my example was not good ... what I meant ...
```java
interface A {
int someField();

@ToString
default String asString() {
// impl
}
}

interface B extends A {
int anotherField();

@ToString
@Override
default String asString() {
// another impl
}
}
```

Since `B` defines another field, it wants to override the `toString()` method of `A` (as it will likely not print the extra field `B` adds ...).

Reply:
Thanks a lot for testing - I see ... OK for me like this.

The only problem I see if two parent interfaces both define an `@ToString` method (potentially with different names) - then you can't extend them both I think ...
One alternative I could see is to do a breadth-first type search: Look for `@ToString` on the actual interface (use it if found), then on any direct parents (use if 1 found, throw if more than 1), then proceed to any parents of the parents etc.
Like this, in the situation above, you could get out of the situation by defining an own `@ToString` method on the top level interface.

But that's probably anyway an extreme edge case, so it's probably not necessary ...

Reply:
YEs, in this case, IT would not be allowed, as there are 2 methods... Indeed, I agree this is a very edge case and not obvious how it should be resolved...

Reply:
So let's leave it for the time being ;-)